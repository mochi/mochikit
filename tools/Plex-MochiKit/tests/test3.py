import Test
import sys
from Plex import *

letter = Range("AZaz") | Any("_")
digit = Range("09")
space = Any(" \t\n")

ident = Seq(letter, Rep(Alt(letter, digit)))
number = Seq(digit, Rep(digit))
punct = Any("*()-+=[]{};:<>,./")
spaces = Seq(space, Rep(space))
resword = Str("program", "begin", "end", "repeat", "until")

lex = Lexicon([
  (resword, TEXT),
  (ident, 'ident'),
  (number, 'num'),
  (punct, TEXT),
  (spaces, IGNORE)
],
debug = Test.debug,
timings = sys.stderr
)

Test.run(lex, "test3", trace = 0)


