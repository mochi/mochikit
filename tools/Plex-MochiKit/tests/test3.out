test3.in   1   0 program    'program'
test3.in   1   8 ident      'furbie'
test3.in   1  14 (          '('
test3.in   1  15 ident      'input'
test3.in   1  20 ,          ','
test3.in   1  22 ident      'output'
test3.in   1  28 ,          ','
test3.in   1  30 ident      'throughput'
test3.in   1  40 )          ')'
test3.in   1  41 ;          ';'
test3.in   2   0 begin      'begin'
test3.in   3   3 repeat     'repeat'
test3.in   4   5 ident      'make'
test3.in   4   9 (          '('
test3.in   4  10 ident      'cute_noises'
test3.in   4  21 )          ')'
test3.in   4  22 ;          ';'
test3.in   5   5 ident      'have_flat_battery'
test3.in   5  22 ;          ';'
test3.in   6   3 until      'until'
test3.in   6   9 ident      'owner_is'
test3.in   6  17 (          '('
test3.in   6  18 ident      'fed_up'
test3.in   6  24 )          ')'
test3.in   6  25 ;          ';'
test3.in   7   0 end        'end'
test3.in   7   3 .          '.'
test3.in  10   0 None       ''
